TheHyruleCastAdventureGame.js

```diff
@@ -0,0 +1,536 @@
+  )
+);
+console.log(
+  chalk.redBright.italic(
+    `Be careful of deadends and ways that sends you flying out of the castle.`
+  )
+);
+
+const game = () => {
+  console.log(
+    chalk.magenta.bold(`I hope you're brave enough to play ${nameOfPlayer}.`)
+  );
+  if (readlineSync.keyInYNStrict(chalk.magenta.bold(`Are you ready ?`))) {
+    startGame();
+  } else {
+    leaveGame();
+  }
+};
+
+const startGame = () => {
+  console.log(chalk.redBright.italic(`Ok ${nameOfPlayer}. Let's start!`));
+  console.log(
+    chalk.redBright.italic(
+      `You got teleported from central Hyrule to entrance of the castle`
+    )
+  );
+  console.log(
+    chalk.redBright.italic(
+      `You're standing in front of what appears to be the big entrance of Hyrule Castle.`
+    )
+  );
+  console.log(
+    chalk.redBright.italic(
+      `You walk in, a Korok appear oferring you a Mighty Elixir to boost your attack and a Tough Elixir to help you in the journey`
+    )
+  );
+  console.log(chalk.redBright.italic(`Your attack went up by + 50`));
+  console.log(chalk.redBright.italic(`Your defense went up by +50`));
+  console.log(
+    chalk.redBright.italic(`There's two huge stairs in front of you.`)
+  );
+
+  choiceOne = readlineSync.question(
+    chalk.magenta.bold(`${nameOfPlayer} Do you wanna go left or right ?\n`)
+  );
+  choiceOne = choiceOne.trim().toLowerCase();
+  if (choiceOne === "left") {
+    console.log(
+      chalk.redBright.italic(
+        `You went up the left stairs and entered the guards' chamber`
+      )
+    );
+    console.log(
+      chalk.redBright.italic(`Inside the chamber, there's a treasure chest.`)
+    );
+    chamberChest();
+  } else if (choiceOne === "right") {
+    console.log(
+      chalk.redBright.italic(
+        `You walked up the right stairs and entered teh dinning hall.`
+      )
+    );
+    console.log(
+      chalk.redBright.italic(
+        `Inside the dining hall, you found a few powerfull dishes that you ate.`
+      )
+    );
+    console.log(
+      chalk.redBright.italic(
+        `You're full now, your attack and defense went up.`
+      )
+    );
+    console.log(chalk.redBright.italic(`There is 3 doors to leave the room.`));
+    console.log(
+      chalk.redBright.italic(
+        `One to the left, one to the right and one forward`
+      )
+    );
+    dinningHallDoors();
+  }
+};
+
+const dinningHallDoors = () => {
+  doorChoiceOne = readlineSync.question(
+    chalk.magenta.bold(`Which door do you want to go through ?\n`)
+  );
+  doorChoiceOne = doorChoiceOne.trim().toLowerCase();
+  if (doorChoiceOne === "left") {
+    dinningHallLeftDoor();
+  } else if (doorChoiceOne === "right") {
+    dinningHallRightDoor();
+  } else if (doorChoiceOne === "forward") {
+    dinningHallForwardDoor();
+  } else {
+    console.log(chalk.redBright.italic(`Please choose a correct answer`));
+    dinningHallDoors();
+  }
```

Comment:
This is a some good logic to place inside of a while loop, instead of calling dinningHallDoors, you can repeat the same lines of code if the doorChoice is invalid.

Reply:
changed this one